### ISSUE TITLE
Enlarge images when users click on them

### ISSUE THREAD

**Issue by ankitm123** [jenkins-x/jx-docs#3089]
**This is a Feature Request**

**What would you like to be added**
I would like to have a zoom in feature for images in the doc site, just like medium. So, when a user hovers over an image, it allows an user to click on the image to enlarge it.

**Why is this needed**
The text in the maturity matrix for jenkins X post 3.0 alpha really looks small, and is hard for readers to see.
https://jenkins-x.io/docs/v3/about/maturity-matrix/

**Comments**
There is a js library which does this: https://github.com/francoischalifour/medium-zoom. A good discussion on how to implement it in hugo can be found here: https://discourse.gohugo.io/t/image-js-zoom-script/27325/5

**Comment by khushishikhu** [comment]
may i work on this

**Comment by ankitm123** [comment]
@khushishikhu Awesome, please assign it to urself, by typing:
```
/assign
```

**Comment by khushishikhu** [comment]
@ankitm123  can you please provide me any reference of how can i setup this repository locally

**Comment by khushishikhu** [comment]
@ankitm123  can you please give me any reference for setting up this repository locally

**Comment by borntorock** [comment]
@khushishikhu you could refer to the instructions mentioned in [README.md](https://github.com/jenkins-x/jx-docs/blob/master/README.md). Feel free to ping in case you notice any step is not clear or  not working.

**Comment by ankitm123** [comment]
> @ankitm123 can you please give me any reference for setting up this repository locally

https://jenkins-x.io/community/documentation/#docker-compose-method

**Comment by khushishikhu** [comment]
![Capture](https://user-images.githubusercontent.com/65439761/96634039-917dc580-1337-11eb-9853-5dafe36cf75b.PNG)
i am just trying to run this repository locally but struck on this warning. i don't know how as i have installed all the prerequisite. please help me to resolve it.

**Comment by ankitm123** [comment]
@khushishikhu Would be better to use docker-compose instead of installing dependencies directly on ur system. Was there any issue with the docker-compose approach (https://github.com/jenkins-x/jx-docs/issues/3089#issuecomment-712107687)?

**Comment by khushishikhu** [comment]
> @khushishikhu Would be better to use docker-compose instead of installing dependencies directly on ur system. Was there any issue with the docker-compose approach ([#3089 (comment)](https://github.com/jenkins-x/jx-docs/issues/3089#issuecomment-712107687))?

yes @ankitm123  i tried to use docker but then my system hangs a lot so i am just not comfortable with docker. That's why i was installing dependencies directly

**Comment by ankitm123** [comment]
So looking at ur error, it seems you are on a windows machine, and hence `make` does not work.
Any chance you could use windows subsystem for linux? https://docs.microsoft.com/en-us/windows/wsl/install-win10
If not, I have to update the docs for windows contributors (we only have instructions for unix-like systems)

**Comment by khushishikhu** [comment]
> So looking at ur error, it seems you are on a windows machine, and hence `make` does not work.
> Any chance you could use windows subsystem for linux? https://docs.microsoft.com/en-us/windows/wsl/install-win10
> If not, I have to update the docs for windows contributors (we only have instructions for unix-like systems)

yes @ankitm123  i can use ubuntu. can you please give me the instruction link for that.

**Comment by ankitm123** [comment]
https://github.com/jenkins-x/jx-docs#locally, basically `make server` should work for ubuntu, after you install the version of hugo mentioned in the [makefile](https://github.com/jenkins-x/jx-docs/blob/master/Makefile)

**Comment by khushishikhu** [comment]
![Capture](https://user-images.githubusercontent.com/65439761/96977741-1be24700-153b-11eb-9786-4c2c2f33bf04.PNG)
i don't know why but it is not excepting the hugo version `0.76.5`

**Comment by ankitm123** [comment]
We are still on hugo version 0.74.0 (https://github.com/jenkins-x/hugo-extended-docker/blob/master/HUGO_VERSION), probably worth upgrading, but not in this PR (You can always create another PR to upgrade it).

So, for now, install version 0.74.0 (https://github.com/gohugoio/hugo/releases/tag/v0.74.0) of hugo, and run make server again. 
Remove ur exisiting version of hugo. Download the hugo extended version from assets section (https://github.com/gohugoio/hugo/releases/download/v0.74.0/hugo_extended_0.74.0_Linux-64bit.deb)
and install the debian file (https://unix.stackexchange.com/a/159114)

Now that ur using ubuntu, my recommendation would be to see if docker-compose is stable on ur ubuntu system or not. As you can already see, it abstracts away all of the headache associated with installing hugo locally.

**Comment by ankitm123** [comment]
https://github.com/jenkins-x/jx-docs/blob/master/Makefile#L1 -> the makefile does a check against this version of hugo, hence you see the error.

**Comment by khushishikhu** [comment]
![Capture](https://user-images.githubusercontent.com/65439761/96998889-f3197c00-1551-11eb-91df-673de9a39ae8.PNG)
@ankitm123  i was using docker on ubuntu and i guess it is almost done but the server is not running. can you please look at this issue and help me to resolve it. Thank you for your guidance.

**Comment by ankitm123** [comment]
> @ankitm123 i was using docker on ubuntu and i guess it is almost done but the server is not running. can you please look at this issue and help me to resolve it. Thank you for your guidance.

Can you post the output of `docker-compose logs -f server`? Also what do u see when u navigate http://localhost:1313/?

**Comment by khushishikhu** [comment]
![Capture](https://user-images.githubusercontent.com/65439761/97005562-3d9ff600-155c-11eb-9086-f1d1ccd469fc.PNG)
now it's running well. after running this command `docker-compose logs -f server`. Thank you!!

**Comment by ankitm123** [comment]
It was working when u ran `make compose-up`, the log command was to just see the logs (and debug if things are going ok or not
). Happy that ur environment is all set, now u can work on the issue :)

**Comment by anudeep-bpgc** [comment]
Hi @ankitm123 , 
This issue seems to be a great choice to start my journey towards FOSS contribution
The issue is still open, may I work on the same.

**Comment by ankitm123** [comment]
@anudeep-bpgc yes, please go ahead!

**Comment by anudeep-bpgc** [comment]
Hi @ankitm123 , I have raised a PR for this issue, kindly review and update regarding the same

**Comment by anudeep-bpgc** [comment]
Hi @ankitm123 , the issue status still stands open even after the successful merge of #3663 . Can you please check once.
